src/query/block/types.go

```diff
@@ -37,6 +37,8 @@ type Block interface {
 	StepIter() (StepIter, error)
 	// SeriesIter returns a SeriesIterator
 	SeriesIter() (SeriesIter, error)
+	// UpdateMetas updates this blocks metadata, and the metadatas for each series.
+	UpdateMetas(Metadata, []SeriesMeta) error
```

Comment:
this is a bit confusing. Shouldn't the metadatas be locked upon block creation ? The block is immutable so we shouldn't be allowed to change anything